apps/src/templates/teacherHomepage/AnnouncementsCarousel.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import FontAwesome from '../FontAwesome';
+import color from "../../util/color";
+import _ from 'lodash';
+
+const styles = {
+  box: {
+    width: 940,
+    backgroundColor: color.white
+  },
+  arrowIcon: {
+    fontSize: 40,
+    lineHeight: "40px",
+    color: color.white,
+    marginRight: 15,
+    marginLeft: 15
+  },
+  arrowBox: {
+    float: "right",
+    marginTop: 12,
+    marginRight: 25,
+    height: 40,
+    width: 90,
+    borderColor: color.white,
+    borderWidth: 1,
+    borderRadius: 3,
+    borderStyle: "solid"
+  },
+  arrowBoxRight: {
+    marginTop: 25,
+    marginRight: 25
+  },
+};
+
+const AnnouncementsCarousel = React.createClass({
+  propTypes: {
+    children: React.PropTypes.oneOfType([
+      React.PropTypes.node,
+      React.PropTypes.arrayOf(React.PropTypes.node)
+    ])
+  },
+
+  getInitialState() {
+    const announcements = this.props.children;
+    return { previous: announcements.length-1, current: 0, next: 1  };
```

Comment:
I track them so that at any point in time I know which announcement to show if the user clicks the right arrow or which to show if they click the left. Are you suggesting that I only track current? And if they click left, show current -1 and to the right, current +1 (and handle cases for the beginning and end of the array)?